R/merge_sce_list.R

```diff
@@ -263,3 +302,131 @@ prepare_sce_for_merge <- function(sce,
   # return the processed SCE
   return(sce)
 }
+
+
+
+#' Prepare altExps for merge by ensuring that all altExps have the same features.
+#'
+#' For any features in `altexp_features` that are missing from the SCE's altExp,
+#'  add those features in with `NA` counts, and update the rowData slot to match.
+#'  This involves creating a new SCE to replace the current altExp.
+#'
+#' @param sce The SCE object whose altExp should be prepared
+#' @param sce_name The name for this SCE
+#' @param altexp_features Vector of features that should be present in the altExp
+#'
+#' @return An updated SCE object with all altExp features present
+prepare_altexps_for_merge <- function(
+    sce,
+    sce_name,
+    altexp_features) {
+
+  sce_altexp <- altExp(sce)
+
+  # Determine which features are missing
+  missing_features <- altexp_features[!(altexp_features %in% rownames(sce_altexp))]
```

Comment:
This common pattern can be shortened
```suggestion
  missing_features <- setdiff(altexp_features, rownames(sce_altexp))
```